localstack/services/apigateway/helpers.py

```diff
@@ -847,61 +847,57 @@ def is_api_key_required(path_payload: dict) -> bool:
                         return True
         return False
 
-    def create_authorizer(path_payload: dict) -> Optional[Authorizer]:
+    def create_authorizers(security_schemes: dict) -> None:
+
+        for security_scheme_name, security_config in security_schemes.items():
+            aws_apigateway_authorizer = security_config.get(OpenAPIExt.AUTHORIZER, {})
+            if not aws_apigateway_authorizer:
+                continue
+
+            if security_scheme_name in authorizers:
+                continue
+
+            authorizer_type = aws_apigateway_authorizer.get("type", "").upper()
+            # TODO: do we need validation of resources here?
+            authorizer = Authorizer(
+                id=create_resource_id(),
+                name=security_scheme_name,
+                type=authorizer_type,
+                authorizerResultTtlInSeconds=aws_apigateway_authorizer.get(
+                    "authorizerResultTtlInSeconds", 300
+                ),
+            )
+            if provider_arns := aws_apigateway_authorizer.get("providerARNs"):
+                authorizer["providerARNs"] = provider_arns
+            if auth_type := security_config.get(OpenAPIExt.AUTHTYPE):
+                authorizer["authType"] = auth_type
+            if authorizer_uri := aws_apigateway_authorizer.get("authorizerUri"):
+                authorizer["authorizerUri"] = authorizer_uri
+            if authorizer_credentials := aws_apigateway_authorizer.get("authorizerCredentials"):
+                authorizer["authorizerCredentials"] = authorizer_credentials
+            if authorizer_type == "TOKEN":
+                header_name = security_config.get("name")
+                authorizer["identitySource"] = f"method.request.header.{header_name}"
+            elif identity_source := aws_apigateway_authorizer.get("identitySource"):
+                # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-authorizer.html
+                # Applicable for the authorizer of the request and jwt type only
+                authorizer["identitySource"] = identity_source
+            if identity_validation_expression := aws_apigateway_authorizer.get(
+                "identityValidationExpression"
+            ):
+                authorizer["identityValidationExpression"] = identity_validation_expression
+
+            rest_api_container.authorizers[authorizer["id"]] = authorizer
+
+            authorizers[security_scheme_name] = authorizer
```

Comment:
this authorizers dict was initially used as a way to avoid duplicates, but then with the rest_api_container we don't need it anymore, essentially they are tracking the same thing. I believe we can remove line 892.

Reply:
makes sense, thanks!

Reply:
it's still practical to directly access a map of the authorizers names to the authorizers, as the `rest_api_container` maps it to their `id` which are randomly generated. I'm using it with `get_authorizer`, I guess it makes sense to keep it in the scope of the import?